rc/filetype/hare.kak

```diff
@@ -7,8 +7,8 @@ hook global BufCreate .*[.]ha %{
 hook global WinSetOption filetype=hare %{
     require-module hare
     hook window ModeChange pop:insert:.* -group hare-trim-indent hare-trim-indent
-    hook window InsertChar \n -group hare-indent hare-indent-on-new-line
     hook window InsertChar \n -group hare-insert hare-insert-on-new-line
+    hook window InsertChar \n -group hare-indent hare-indent-on-new-line
```

Comment:
I remember that the hook order was relevant for some other reason, and that we are not consistent about that.
IIRC for the more complex file types it's convenient to use a certain hook order, for others it doesn't matter.

Probably the hare module would be better of staying closer to established filetypes. For example, `go.kak` does not have this problem despite using the same hook order

`hare-insert-on-new-line` includes a snippet that does

	# remove trailing whitespace on the above line

Other filetype highlighter seem to have the same feature but implemented in a different way.
That function could probably just be (untested)

    define-command -hidden hare-insert-on-new-line %{ evaluate-commands -draft -itersel %{
        # copy // comments prefix and following white spaces
        try %{ execute-keys -draft <semicolon><c-s>kx s ^\h*\K/{2,}\h* <ret> y<c-o>P<esc> }
    } }